File: test/leetcode/graph/shortest-path-in-binary-matrix.test.ts
import { shortestPathBinaryMatrix } from '../../src/graph/shortest-path-in-binary-matrix';

describe('shortest path in binary matrix', () => {
  test('shortest path in binary matrix - test case 1', async () => {
    const grid = [
      [0, 1],
      [1, 0]
    ];

    expect(shortestPathBinaryMatrix(grid)).toBe(2);
  });

  test('shortest path in binary matrix - test case 2', async () => {
    const grid = [
      [0, 1, 1, 0, 0, 0],
      [0, 1, 0, 1, 1, 0],
      [0, 1, 1, 0, 1, 0],
      [0, 0, 0, 1, 1, 0],
      [1, 1, 1, 1, 1, 0],
      [1, 1, 1, 1, 1, 0]
    ];

    expect(shortestPathBinaryMatrix(grid)).toBe(14);
  });
});

File: test/leetcode/graph/shortest_path_in_binary_matrix_test.py
from leetcode.graph.shortest_path_in_binary_matrix import Solution


soln = Solution()


def test_case_1():
    grid = [[0, 1], [1, 0]]

    assert soln.shortestPathBinaryMatrix(grid) == 2


def test_case_2():
    grid = [
        [0, 1, 1, 0, 0, 0],
        [0, 1, 0, 1, 1, 0],
        [0, 1, 1, 0, 1, 0],
        [0, 0, 0, 1, 1, 0],
        [1, 1, 1, 1, 1, 0],
        [1, 1, 1, 1, 1, 0],
    ]

    assert soln.shortestPathBinaryMatrix(grid) == 14
